sdks/java/io/iceberg/src/test/java/org/apache/beam/sdk/io/iceberg/RecordWriterManagerTest.java

```diff
@@ -739,39 +740,45 @@ public void testColumnSpecificMetricsCollection() throws IOException {
-    assertTrue("Write operation should succeed", writeSuccess);
+    // Write multiple rows
+    Row row1 = Row.withSchema(BEAM_SCHEMA).addValues(1, "aaa", true).build();
+    Row row2 = Row.withSchema(BEAM_SCHEMA).addValues(2, "bbb", false).build();
+    Row row3 = Row.withSchema(BEAM_SCHEMA).addValues(3, "ccc", true).build();
+    assertTrue("Write operation should succeed", writerManager.write(windowedDestination, row1));
+    assertTrue("Write operation should succeed", writerManager.write(windowedDestination, row2));
+    assertTrue("Write operation should succeed", writerManager.write(windowedDestination, row3));
     writerManager.close();
 
     Map<WindowedValue<IcebergDestination>, List<SerializableDataFile>> dataFiles =
         writerManager.getSerializableDataFiles();
     assertFalse("Data files should not be empty", dataFiles.isEmpty());
 
+    // Commit using the same table instance
     for (Map.Entry<WindowedValue<IcebergDestination>, List<SerializableDataFile>> entry :
         dataFiles.entrySet()) {
-      Table tableToCommit = catalog.loadTable(entry.getKey().getValue().getTableIdentifier());
-      AppendFiles appendFiles = tableToCommit.newAppend();
+      AppendFiles appendFiles = table.newAppend();
       for (SerializableDataFile dataFile : entry.getValue()) {
-        appendFiles.appendFile(dataFile.createDataFile(tableToCommit.specs()));
+        appendFiles.appendFile(dataFile.createDataFile(table.specs()));
       }
       appendFiles.commit();
-      tableToCommit.refresh();
     }
+    table.refresh(); // Single refresh after all commits
 
-    // Verify the correct table (where data was written)
-    Table dataTable =
-        catalog.loadTable(
-            TableIdentifier.of("default", "table_testColumnSpecificMetricsCollection"));
-    dataTable.refresh();
-
-    // Verify that a snapshot exists
-    Snapshot snapshot = dataTable.currentSnapshot();
+    // Verify using the same table
+    Snapshot snapshot = table.currentSnapshot();
     assertNotNull("Table should have a snapshot after writing data", snapshot);
 
-    // Verify metrics are collected only for specified columns
-    DataFile dataFile = snapshot.addedDataFiles(dataTable.io()).iterator().next();
-    assertNotNull(dataFile.valueCounts());
+    // Verify metrics are collected for specified columns in one file
+    DataFile dataFile = snapshot.addedDataFiles(table.io()).iterator().next();
+    Map<Integer, Long> valueCounts = dataFile.valueCounts();
+    assertNotNull("Value counts should not be null", valueCounts);
+    assertTrue("Value counts should exist for id (column 1)", valueCounts.containsKey(1));
+    assertEquals("Value count for id should be 1 in this file", 1L, valueCounts.get(1).longValue());
+    assertTrue("Value counts should exist for name (column 2)", valueCounts.containsKey(2));
+    assertEquals(
+        "Value count for name should be 1 in this file", 1L, valueCounts.get(2).longValue());
```

Comment:
We're writing 3 records, so shouldn't these value counts equal 3?

Reply:
I think it's because we're creating the table with a partition spec above, so this test is actually writing 3 different data files, with one row in each.

Can we remove the partition spec so we only have one data file?